ihp-ide/IHP/SchemaCompiler.hs

```diff
@@ -903,14 +1043,48 @@ compileSetFieldInstances table@(CreateTable { name, columns }) = unlines (map co
                     | otherwise = name'
 
 compileUpdateFieldInstances :: (?schema :: Schema) => CreateTable -> Text
-compileUpdateFieldInstances table@(CreateTable { name, columns }) = unlines (map compileSetField (dataFields table))
+compileUpdateFieldInstances table@(CreateTable { name, columns, inherits }) = unlines (map compileSetField (dataFields table))
     where
         modelName = tableNameToModelName name
-        typeArgs = dataTypeArguments table
-        compileSetField (name, fieldType) = "instance UpdateField " <> tshow name <> " (" <> compileTypePattern table <>  ") (" <> compileTypePattern' name  <> ") " <> valueTypeA <> " " <> valueTypeB <> " where\n    {-# INLINE updateField #-}\n    updateField newValue (" <> compileDataTypePattern table <> ") = " <> modelName <> " " <> (unwords (map compileAttribute (table |> dataFields |> map fst)))
+        -- @todo: Find a better way.
+        colName = modelName |> pluralize |> Text.toLower
+
+        currentTypeArguments = dataTypeArguments table
+
+
+        parentTypeArguments :: [Text]
+        parentTypeArguments =
+            case inherits of
+                Nothing -> []
+                Just parentTable ->
+                    let parentTableDef = findTableByName parentTable
+                    in parentTableDef
+                            |> maybe [] (dataTypeArguments . (.unsafeGetCreateTable))
+                            -- We remove ref to own table (e.g. `post_revisions` table should not have postRevisions)
+                            |> filter (\fieldName -> Text.toLower fieldName /= colName)
+
+
+        allTypeArguments = currentTypeArguments <> parentTypeArguments
+
+        currentDataFields = dataFields table |> filter (\(fieldName, _) -> fieldName /= "meta")
+
+        parentDataFields = case inherits of
+            Nothing -> []
+            Just parentTable ->
+                let parentTableDef = findTableByName parentTable
+                in parentTableDef
+                        |> maybe [] (dataFields . (.unsafeGetCreateTable))
+                        -- We remove ref to own table (e.g. `post_revisions` table should not have postRevisions)
+                        -- @todo: Check name of `id` column.
+                        |> filter (\(fieldName, _) -> fieldName /= "meta" && Text.toLower fieldName /= colName && fieldName /= "id")
+
+
+        allDateFields = currentDataFields <> parentDataFields <> [("meta", "MetaBag")]
+
+        compileSetField (name, fieldType) = "instance UpdateField " <> tshow name <> " (" <> compileTypePattern table <>  ") (" <> compileTypePattern' name  <> ") " <> valueTypeA <> " " <> valueTypeB <> " where\n    {-# INLINE updateField #-}\n    updateField newValue (" <> compileDataTypePattern table <> ") = " <> modelName <> " " <> (unwords (map compileAttribute (allDateFields |> map fst)))
```

Comment:
@mpscholten At this point I'm just trying to get things to compile. It would require code cleanup (but open for early review, if you see some evident things).

Right now I have this error - and I admit those types are confusing me :sweat_smile: 

![image](https://github.com/user-attachments/assets/ae780605-cabb-4a88-bcb1-f977481f9a23)

```haskell
instance UpdateField "postId" (PostRevision' postId userId) (PostRevision' postId' userId) postId postId' where
    {-# INLINE updateField #-}
    updateField newValue (PostRevision id postId title body userId meta) = PostRevision id newValue title body userId (meta { touchedFields = "postId" : touchedFields meta })
```

Reply:
Ok, I think I know why, because the order of fields is not consistent, as previously defined:

```
data PostRevision' postId userId = PostRevision {title :: Text, body :: Text, userId :: userId, id :: (Id' "post_revisions"), postId :: postId, meta :: MetaBag} deriving (Eq, Show)
```

I'll fix that